### DISCUSSION TITLE
Cursor blink doesn't work

### DISCUSSION THREAD

**Discussion by JuanZoran** [wezterm/wezterm#3512] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/107862700/231634927-4b170ee0-9ccb-4bc6-8473-e4bffe8c2af4.png)
I try to use this minimal configuration to debug this issue, but I found that it doesn't work ...

**Comment by nekowinston** [comment]
100 in this case are milliseconds - the easing likely makes this hard to see. Try
```lua
c.cursor_blink_rate = 100
c.cursor_blink_ease_in = "Constant"
c.cursor_blink_ease_out = "Constant"
```

**Reply by JuanZoran** [reply]
![image](https://user-images.githubusercontent.com/107862700/231636950-cd8943cc-d968-4518-b0cd-62b8ccf7987e.png)
I just use that option, but it still doesn't work at all...

**Reply by nekowinston** [reply]
If you expect to see a change inside `nvim`, have a look at [`:help 'guicursor'`](https://neovim.io/doc/user/options.html#'guicursor')

**Reply by JuanZoran** [reply]
I had read about that option, but I found that's not because of neovim option...
even I was in the command line, the cursor is still not blinking...

https://user-images.githubusercontent.com/107862700/231638845-a2664caa-4263-419e-9f5b-7cd784265135.mp4

**Reply by nekowinston** [reply]
Does it blink if you skip loading your zsh config? `zsh -d -f -i`

**Reply by JuanZoran** [reply]
Yes! It works, but I don't why the cursor blinking works when I use the kitty terminal but not in wezterm....
when I use wezterm, I should try to use `zsh -d -f -i`, then the cursor will blink
but when I use kitty, it just works

**Reply by nekowinston** [reply]
I'm glad someone else can replicate this.
I assumed that the Vi-mode plugin I use with zsh caused this, but what's curious is that `zsh -v` or `zsh -v -i` also makes the cursor blink.

**Reply by JuanZoran** [reply]
Actually I am using zsh-vi-mode plugin,  did u find any solution for that? I really enjoy the cursor blinking animation ... 

> by the way, I think the best command line animation is the **smooth cursor** and **smooth cartet** animation in vscode... 
> I don't know why there is not a terminator that implements that

**Reply by nekowinston** [reply, marked as answer]
I just figured out that the [Vi-mode](https://github.com/jeffreytse/zsh-vi-mode/blob/master/README.md#custom-cursor-style) indeed caused this on my end, I added `ZVM_CURSOR_STYLE_ENABLED=false` to my early `.zshenv`, which fixed this for me.
https://github.com/nekowinston/dotfiles/commit/79a5f830b46d5df915db338e9dd92135a1d9a4b9

Edit: zshenv actually

**Reply by JuanZoran** [reply]
Thanks a lot, I just add these to my .zshrc
now the cursor blinking works well
```zsh
export ZVM_CURSOR_STYLE_ENABLED=false
function zle-keymap-select {
    if [[ ${KEYMAP} == vicmd ]] || [[ $1 = 'block' ]]; then
        echo -ne '\e[1 q'
    elif [[ ${KEYMAP} == main ]] || [[ ${KEYMAP} == viins ]] || [[ ${KEYMAP} = '' ]] || [[ $1 = 'beam' ]]; then
        echo -ne '\e[5 q'
    fi
}

zle -N zle-keymap-select

_fix_cursor() {
    echo -ne '\e[5 q'
}
precmd_functions+=(_fix_cursor)
```

but I don't know how the `text_blink_rate` and `text_blink_ease_in` make sense

**Reply by nekowinston** [reply]
Those are for blinking text that's printed to the terminal: `echo  -e "\033[5mBlinking Text\033[0m"`

**Reply by JuanZoran** [reply]
Thanks! Does wezterm have any option can implement the vscode `Cursor Smooth Caret Animation`

**Reply by nekowinston** [reply]
No. See the issue and mentions of the issue in https://github.com/wez/wezterm/issues/1432 / https://github.com/wez/wezterm/discussions/1136

**Reply by JuanZoran** [reply]
I just found that I can set the insert mode cursor as "ZVM_CURSOR_BLINKING_BEAM" and I don't need to disable the ZVM_CURSOR_STYLE_ENABLED, like this 
![image](https://user-images.githubusercontent.com/107862700/231664629-9bc26001-ceea-4ab2-b396-a1724fdcf186.png)
